modules/network/networker.go

```diff
@@ -1,27 +1,200 @@
+
+func (n *networker) GetNetResource(id string) (modules.NetResource, error) {
+	// TODO check signature
+	return n.netResAlloc.Get(id)
+}
+
+func (n *networker) ApplyNetResource(netID modules.NetID, resource modules.NetResource) error {
+	return applyNetResource(n.storageDir, netID, resource)
+}
+
+func applyNetResource(storageDir string, netID modules.NetID, netRes modules.NetResource) error {
+	network := string(netID)
+	if err := createNetwork(network); err != nil {
+		return err
+	}
+
+	storage := filepath.Join(storageDir, network)
+	if err := configureWG(storage, network, netRes); err != nil {
+		return err
+	}
+	return nil
+}
+
+// createNetwork creates a network namespace and a bridge
+// and a wireguard interface and then move it interface inside
+// the net namespace
+func createNetwork(name string) error {
+	var (
+		netnsName  = netnsName(name)
+		bridgeName = bridgeName(name)
+		wgName     = wgName(name)
+	)
+
+	if !namespace.Exists(netnsName) {
+		log.Info().Str("name", netnsName).Msg("create network namespace")
+		_, err := namespace.Create(netnsName)
+		if err != nil {
+			return err
+		}
+
+		log.Info().Str("name", wgName).Msg("create wireguard interface")
+		wg, err := wireguard.New(wgName)
+		if err != nil {
+			return err
+		}
+
+		log.Info().
+			Str("wg", wgName).
+			Str("namespace", netnsName).
+			Msg("move wireguard into network namespace")
+		if err := namespace.SetLink(wg, netnsName); err != nil {
+			return err
+		}
+	}
+
+	if !bridge.Exists(bridgeName) {
+		if _, err := bridge.New(bridgeName); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func deleteNetwork(network string) error {
+	if err := bridge.Delete(bridgeName(network)); err != nil {
+		return err
+	}
+	return namespace.Delete(netnsName(network))
+}
+
+func configureWG(storageDir, network string, netRes modules.NetResource) error {
+	var (
+		netnsName   = netnsName(network)
+		wgName      = wgName(network)
+		storagePath = filepath.Join(storageDir, network)
+		key         wgtypes.Key
+		err         error
+	)
+
+	key, err = wireguard.LoadKey(storagePath)
+	if err != nil {
+		key, err = wireguard.GenerateKey(storagePath)
+		if err != nil {
+			return err
+		}
+	}
+
+	// enter container net ns
+	nsCtx := namespace.NSContext{}
+	if err := nsCtx.Enter(netnsName); err != nil {
+		return err
+	}
+
+	wg, err := wireguard.GetByName(wgName)
+	if err != nil {
+		return err
+	}
+
+	// configure wg iface
+	peers := make([]wireguard.Peer, len(netRes.Connected))
+	for i, connected := range netRes.Connected {
+		if connected.Type != modules.ConnTypeWireguard {
+			continue
+		}
+
+		conn := connected.Connection
+
+		var endpoint string
+		if conn.Peer.To16() != nil {
+			endpoint = fmt.Sprintf("[%s]:%d", conn.Peer.String(), conn.PeerPort)
+		} else {
+			endpoint = fmt.Sprintf("%s:%d", conn.Peer.String(), conn.PeerPort)
+		}
+
+		peers[i] = wireguard.Peer{
+			PublicKey:  conn.Key,
+			Endpoint:   endpoint,
+			AllowedIPs: []string{"0.0.0.0/0"},
```

Comment:
AllowedIPs has to contain: the fe80::/ of the WG, the Prefix of the Network Resource on the bridge, an IPv4 LL addr and an IPv4 subnet on the bridge. No peer can heve the same IP (so certainly not 0/0)

Reply:
But don't you need this for the wg interface to be able to forward all traffic in/out ? I though that without this the interface can actually only reach other peer. But the exit not cannot send public traffic to it

Reply:
Hmm.. 
I'll setup testnet.sh to reflect that properly